GenericsHomework/Node.cs

```diff
@@ -0,0 +1,63 @@
+﻿namespace GenericsHomework;
+public class Node<TVal> where TVal : notnull
+{
+
+    public Node(TVal val)
+    {
+        Val = val ?? throw new ArgumentNullException($"Node<{typeof(TVal)}> value cannot be null");
+        Next = this;
+        
+    }
+
+    public TVal Val{ get; set; }
+    public Node<TVal> Next { get; private set; }
+   
+
+    public void Append(TVal node)
+    {
+        if(Exists(node))
+            throw new ArgumentException("This value already exists. Values must be unique.");
+
+        Node<TVal> node1 = new(node);
+        if(Next == this)
+        {
+            Next = node1;
+            node1.Next = this;
+        }
+        else
+        {
+            node1.Next = Next;
+            Next = node1;
+        }        
```

Comment:
```suggestion
            Node<T> newNode = new(node);
            newNode.Next = Next;
            Next = newNode;
            return newNode; 
```
Another way to do this